html/modules/custom/reliefweb_analytics/src/Command/ReliefwebMostReadCommand.php

```diff
@@ -0,0 +1,594 @@
+   */
+  public function fetchGa4DataCombined(array $parameters) {
+    $results = [];
+
+    try {
+      $start = microtime(TRUE);
+      $response = $this->getGa4Client()->runReport($parameters);
+
+      // Make sure it takes at least a second.
+      $end = microtime(TRUE);
+      if ($end - $start < 1) {
+        usleep($end - $start);
+      }
+    }
+    catch (ApiException $exception) {
+      if ($exception->getStatus() == 'RESOURCE_EXHAUSTED') {
+        $this->logger()->warning('Rate limit hit.');
+        $this->logger()->warning('Google exception: ' . $exception->getMessage());
+      }
+      else {
+        $this->logger()->error('Google exception: ' . $exception->getMessage());
+      }
+      exit();
+    }
+    catch (\Exception $exception) {
+      $this->logger()->error('Exception: ' . $exception->getMessage());
+      exit();
+    }
+
+    // Log quota.
+    $quota = $response->getPropertyQuota();
+    $this->logger()->notice(strtr('Day: @d, hour: @h, errors/hour: @e, threshold: @t', [
+      '@d' => $quota->getTokensPerDay()->getConsumed() . '/' . $quota->getTokensPerDay()->getRemaining(),
+      '@h' => $quota->getTokensPerHour()->getConsumed() . '/' . $quota->getTokensPerHour()->getRemaining(),
+      '@e' => $quota->getServerErrorsPerProjectPerHour()->getConsumed() . '/' . $quota->getServerErrorsPerProjectPerHour()->getRemaining(),
+      '@t' => $quota->getPotentiallyThresholdedRequestsPerHour()->getConsumed() . '/' . $quota->getPotentiallyThresholdedRequestsPerHour()->getRemaining(),
+    ]));
+
+    foreach ($response->getRows() as $row) {
+      if (isset($row->getDimensionValues()[1]) && !empty($row->getDimensionValues()[1]->getValue()) && $row->getDimensionValues()[1]->getValue() !== '(not set)') {
+        // Expand $row->getDimensionValues()[1].
+        $parts = explode(',', $row->getDimensionValues()[1]->getValue());
+        foreach ($parts as $part) {
+          $part = trim($part);
+          if (!isset($results[$part])) {
+            $results[$part] = [];
+          }
+          if (count($results[$part]) < 5) {
+            $results[$part][] = $row->getDimensionValues()[0]->getValue();
+          }
+        }
+      }
+    }
+
+    return $results;
+  }
+
+  /**
+   * Our logger.
+   */
+  protected function getLogger(): LoggerChannelInterface {
+    return $this->loggerFactory->get('reliefweb_analytics');
+  }
+
+  /**
+   * Common payload.
+   */
+  protected function buildPayload($filter = NULL) {
+    $expressions = [
+      new FilterExpression([
+        'filter' => new Filter([
+          'field_name' => 'customEvent:content_type',
+          'string_filter' => new StringFilter([
+            'value' => 'Report',
+          ]),
+        ]),
+      ]),
+      new FilterExpression([
+        'filter' => new Filter([
+          'field_name' => 'pagePath',
+          'string_filter' => new StringFilter([
+            'match_type' => MatchType::BEGINS_WITH,
+            'value' => '/report',
```

Comment:
@attiks Sorry, I just noticed that map reports have their URL path prefix with `/map` not `/report` (ex: https://reliefweb.int/map/mongolia/mongolia-drought-map-ten-day-period-starting-11-june-2022-enmn) so the filter should check both. Hopefully that will not increase much the number of tokens that the query uses.

Reply:
Or we remove it, there's already a filter on content type?

Reply:
The `/report` filter was there initially because there were some `/node/1234566/edit` (from editors) URLs "polluting" the results but it's probably rare enough to not care.

So, yes, let's remove it and see how it goes. That may reduce the number of tokens required.

Reply:
Removed